lib/ansible/modules/system/known_hosts.py

```diff
@@ -98,16 +101,19 @@ def enforce_state(module, params):
     # Find the ssh-keygen binary
     sshkeygen = module.get_bin_path("ssh-keygen", True)
 
+    if key is None and state != "absent":
+        module.fail_json(msg="No key specified when adding a host")
+
+    if key is not None and hash_host is True:
```

Comment:
`if key is not None and hash_host:` could be used too.

Reply:
in that sense `if key and hash_host` would probably do it too, I opted for this variant because the rest of the code uses it like this too.

Reply:
Indeed `hash_host_key` method will fail if `key` is an empty string. The previous test should be modified too.